syntax = "proto3";

import "simpletest.proto";

package qtprotobufnamespace.tests;

service TestService {
  rpc testMethod(SimpleStringMessage) returns (SimpleStringMessage) {}
  rpc testMethodServerStream(SimpleStringMessage) returns (stream SimpleStringMessage) {}
  rpc testMethodClientStream(stream SimpleStringMessage) returns (SimpleStringMessage) {}
  rpc testMethodBiStream(stream SimpleStringMessage) returns (stream SimpleStringMessage) {}
  rpc testMethodBlobServerStream(BlobMessage) returns (stream BlobMessage) {}
  rpc testMethodStatusMessage(SimpleStringMessage) returns (SimpleStringMessage) {}
  rpc testMethodNonCompatibleArgRet(SimpleIntMessage) returns (SimpleStringMessage) {}
} 
